lzss/decompress.go

```diff
@@ -70,60 +69,154 @@ func Decompress(data, dict []byte) (d []byte, err error) {
 	return out.Bytes(), nil
 }
 
-// ReadIntoStream reads the compressed data into a stream
-// the stream is not padded with zeros as one obtained by a naive call to compress.NewStream may be
-// Deprecated: use Compressor.Stream() instead
-func ReadIntoStream(data, dict []byte, level Level) (compress.Stream, error) {
+type CompressionPhrase struct {
+	Type              byte
+	Length            int
+	ReferenceAddress  int
+	StartDecompressed int
+	StartCompressed   int
+	Content           []byte
+}
+
+type CompressionPhrases []CompressionPhrase
+
+func CompressedStreamInfo(c, dict []byte) CompressionPhrases {
+	in := bitio.NewReader(bytes.NewReader(c))
 
-	out, err := compress.NewStream(data, uint8(level))
+	// parse header
+	var header Header
+	sizeHeader, err := header.ReadFrom(in)
 	if err != nil {
```

Comment:
maybe modify signature here to return `CompressionPhrases, error` and propagate the error